Allowing admin users to change users email

DieselTech opened the pull request:
# Changed
- Changed: Allowing admins to edit user profiles to update their email addresses. (Closes https://github.com/Kareadita/Kavita/issues/2647)

DieselTech committed:
Allowing admin users to change users email

fixes https://github.com/Kareadita/Kavita/issues/2647

majora2007 commented on API/Controllers/AccountController.cs:
```diff
@@ -508,6 +508,19 @@ public async Task<ActionResult> UpdateAccount(UpdateUserDto dto)
+            // Validate email change
+            var emailValidationErrors = await _accountService.ValidateEmail(dto.Email);
+            if (emailValidationErrors.Any()) return BadRequest(await _localizationService.Translate(User.GetUserId(), "email-taken"));
+            user.Email = dto.Email;
```
You need a new line after the if statement to help the code be easier to digest.

majora2007 commented on API/Controllers/AccountController.cs:
```diff
@@ -508,6 +508,19 @@ public async Task<ActionResult> UpdateAccount(UpdateUserDto dto)
+            user.Email = dto.Email;
+            user.EmailConfirmed = false;
+            user.NormalizedEmail = _userManager.NormalizeEmail(dto.Email);
+            user.ConfirmationToken = await _userManager.GenerateEmailConfirmationTokenAsync(user);
```
You generate a confirmation token, but I don't see you triggering an email for the end user to confirm and get rid of that pending status.

majora2007 reviewed (changes requested):
Just did a preliminary review. I don't see any of the email code to trigger a link so the user can confirm the email change.

majora2007 commented:
Not sure if you still want to work on this to get it in?

DieselTech committed:
Update AccountController.cs

Added new line as requested
Removed the confirmation token since if the admin is changing the users email, it should be a force change.
Set the `EmailConfirmed` status to `True`, again since the admin is the one pushing it through.

DieselTech commented:
Changes made. Got rid of the email confirmation step because it is an admin making the change. That should force change it and not rely on the confirmation flow. Not sure what I was thinking before.

majora2007 commented:
I'm going to implement this myself, it's going to be much easier. I'll dm you the correct way to implement this.

majora2007 commented:
Here is the correct way to implement:
https://github.com/Kareadita/Kavita/pull/3446/commits/d522bccf86cb11efd1b59e380bbbc446c2c6ba35
